pkg/rest/server.go

```diff
@@ -49,6 +49,7 @@ func (s *Server) Start() {
 			// as main() doesn't wait for this goroutine to stop. don't use
 			// code with race conditions like these for production. see post
 			// comments below on more discussion on how to handle this.
+			//Also don't know what to do with this one
 			log.Fatalf("ListenAndServe(): %s", err)
```

Comment:
```
// Start the server listening for incoming requests.
func (s *Server) Start() error {
	serverLog.Info("Starting server on %+v\n", s.server.Addr)
	return s.server.ListenAndServe()
}
```
and then in the caller
```
go func(){
	if err := ds.server.Start(); err!=nil {
		// error log
	}
}()
	 
```

Reply:
This one isn't really clear to me... I don't really understand the flow of what is calling what. I'll check again tomorrow and ping you if I'm still in doubt

Reply:
yep we can do a quick call.

Reply:
Whatś  the correct syntax for this though?
```
// Start the server listening for incoming requests.
func (s *Server) Start() error {
	return go func() error {
		serverLog.Info("Starting server on %+v\n", s.server.Addr)
		return s.server.ListenAndServe()
	}()
}
```